schema/deploy/roles/materialized-view-refresher/grants.sql

```diff
@@ -0,0 +1,44 @@
+-- Deploy seattleflu/id3c-customizations:roles/materialized-view-refresher/grants to pg
+-- requires: roles/materialized-view-refresher/create
+
+begin;
+
+revoke all on database :"DBNAME" from "materialized-view-refresher";
+revoke all on schema receiving, warehouse, shipping from "materialized-view-refresher";
+revoke all on all tables in schema receiving, warehouse, shipping from "materialized-view-refresher";
+
+grant connect on database :"DBNAME" to "materialized-view-refresher";
+
+-- Regular users cannot refresh materialized views because only the view
+-- owner has permission to refresh them.
+-- Create a function that runs in the security context of its owner so that
+-- other users can refresh the view.
+create or replace function public.refresh_materialized_view(view_name text)
+    returns void
+    security definer
+    language plpgsql as $$
+        declare
+            sql text;
+            message text;
+        begin
+            sql := 'refresh materialized view concurrently ' || view_name;
+            execute sql;
```

Comment:
Because `execute` allows multiple statements to be run and the view name is not quoted, there's SQL injection vulnerability here which would let anyone with `materialized-view-refresher` run SQL as the superuser (since `postgres` is the function owner). For example, consider:

    select refresh_materialized_view('shipping.fhir_questionnaire_responses_v1; drop database production;');

:grinning: 

To fix this, we can quote `view_name` as an identifier, but will first have to change the function signature to take the schema and name separately:

    execute format('refresh materialized view concurrently %I.%I', view_schema, view_name);

Reply:
😬 I have to work on thinking more defensively, but cool I didn't know about `%I`